pkg/cache/v3/simple.go

```
@@ -210,6 +211,28 @@ func (cache *snapshotCache) SetSnapshot(node string, snapshot Snapshot) error {
 				delete(info.watches, id)
 			}
 		}
+
+		// this should always pass since we initialize on snapshot creation
```

Let's remove the check then? Avoids having to debug silent failures